src/nodes/utils.ts

```diff
@@ -647,10 +647,11 @@ async function verifyClientCertificateChain(
     certChain.push(cert);
   }
   const now = new Date();
+  let leafCert = true;
   for (let certIndex = 0; certIndex < certChain.length; certIndex++) {
     const cert = certChain[certIndex];
     const certNext = certChain[certIndex + 1];
-    if (now < cert.notBefore || now > cert.notAfter) {
+    if (leafCert && (now < cert.notBefore || now > cert.notAfter)) {
```

Comment:
This is a bit confusing when it's set to true and then checked. The order of switching booleans may be clearer if it was the other way around.

Reply:
As in, started as false and had a different name?